projects/bms_carrier/test/test_passive_balance.c

```diff
@@ -0,0 +1,120 @@
+// Test sequence for passive balancing module
+#include "passive_balance.h"
+
+#include "cell_sense.h"
+#include "ltc_afe.h"
+#include "ms_test_helpers.h"
+
+#include "log.h"
+
+static uint16_t s_cell_balanced = 0;
+static bool s_cell_discharge = false;
+
+StatusCode TEST_MOCK(ltc_afe_toggle_cell_discharge)(LtcAfeStorage *afe, uint16_t cell,
+                                                    bool discharge) {
+  s_cell_balanced = cell;
+  s_cell_discharge = discharge;
+  return STATUS_CODE_OK;
+}
+
+// To store voltages for test
+static uint16_t s_test_voltages[NUM_TOTAL_CELLS];
```

Comment:
Might wanna comment that this number is expected to be at least 4.

or ... you can set a local max cell define that is set to be the max of TOTAL_CELLS and 4.